From 85982b69871eeebb00afac1ba3039e45e7ab3a1f Mon Sep 17 00:00:00 2001
From: Angela Fong <angela.fong@joyent.com>
Date: Tue, 13 Mar 2018 22:23:24 -0700
Subject: [PATCH] ADMINUI-2401 Want support for bhyve brand Reviewed by:
 Marsell Kukuljevic <marsell@joyent.com> Approved by: Marsell Kukuljevic
 <marsell@joyent.com>

---
 www/js/components/pages/images/list.jsx | 2 +-
 www/js/components/pages/vm/index.jsx    | 6 +++---
 www/js/tpl/provision-vm.hbs             | 1 +
 www/js/views/provision-vm.js            | 8 ++++----
 4 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/www/js/components/pages/images/list.jsx b/www/js/components/pages/images/list.jsx
index 698fcc76..84392ff8 100644
--- a/www/js/components/pages/images/list.jsx
+++ b/www/js/components/pages/images/list.jsx
@@ -14,7 +14,7 @@ var ImagesList = React.createClass({
         var href = '/images/' + img.uuid;
         var model = new Image(img);
         var publishDate = moment(img.published_at).utc().format('MM/DD/YYYY');
-        var os = img.type === 'zvol' ? ('KVM, ' + img.os) : img.os;
+        var os = img.type === 'zvol' ? ('HVM, ' + img.os) : img.os;
 
         return <tr key={img.uuid}>
             <td className="state">
diff --git a/www/js/components/pages/vm/index.jsx b/www/js/components/pages/vm/index.jsx
index 3043181a..196cea95 100644
--- a/www/js/components/pages/vm/index.jsx
+++ b/www/js/components/pages/vm/index.jsx
@@ -110,7 +110,7 @@ var VMPage = React.createClass({
         var vm = this.state.vm;
 
         var quota = 0;
-        if (vm.brand === 'kvm') {
+        if (vm.brand === 'kvm' || vm.brand === 'bhyve') {
             vm.disks.forEach(function (k) {
                 if (k.size && typeof(k.size) === 'number' && k.size > 0) {
                     quota = quota + (k.size/1024); // disk.size is in mib
@@ -158,7 +158,7 @@ var VMPage = React.createClass({
                         <li className="divider"></li>
                         <li><a onClick={this._handleRenameVm} className="rename">Rename Alias</a></li>
 
-                        { vm.brand !== 'kvm' ?
+                        { vm.brand !== 'kvm' && vm.brand !== 'bhyve' ?
                             [
                                 <li key="1" className="divider"></li>,
                                 <li key="2"><a onClick={this._handleReprovisionVm} className="reprovision">Reprovision</a></li>,
@@ -200,7 +200,7 @@ var VMPage = React.createClass({
                                 <th>Disk</th>
                                 <td>
                                     <span className="vm-disk-quota">
-                                    {vm.brand === 'kvm' ?
+                                    {vm.brand === 'kvm' || vm.brand === 'bhyve' ?
                                         <div className="vm-disk-quota-kvm">
                                             Total: {quota} GB
                                             {vm.disks && vm.disks.length && vm.disks[0].image_size ?
diff --git a/www/js/tpl/provision-vm.hbs b/www/js/tpl/provision-vm.hbs
index 782ff740..960e6cf8 100644
--- a/www/js/tpl/provision-vm.hbs
+++ b/www/js/tpl/provision-vm.hbs
@@ -66,6 +66,7 @@
                     <option value="joyent">joyent</option>
                     <option value="joyent-minimal">joyent-minimal</option>
                     <option value="kvm">kvm</option>
+                    <option value="bhyve">bhyve</option>
                     <option value="lx">lx</option>
                     <option value="sngl">sngl</option>
                 </select>
diff --git a/www/js/views/provision-vm.js b/www/js/views/provision-vm.js
index aea4bf18..7c425d65 100644
--- a/www/js/views/provision-vm.js
+++ b/www/js/views/provision-vm.js
@@ -335,10 +335,10 @@ var View = Backbone.Marionette.Layout.extend({
                 this.disableBrands('joyent', 'joyent-minimal', 'lx', 'sngl');
             } else if (image.get('type') === 'lx-dataset') {
                 this.setBrand('lx');
-                this.disableBrands('joyent', 'joyent-minimal', 'kvm', 'sngl');
+                this.disableBrands('joyent', 'joyent-minimal', 'kvm', 'bhyve', 'sngl');
             } else if (image.get('type') === 'zone-dataset') {
                 this.setBrand('joyent');
-                this.disableBrands('kvm', 'lx', 'sngl');
+                this.disableBrands('kvm', 'bhyve', 'lx', 'sngl');
             } else {
                 this.disableBrands(false);
             }
@@ -450,7 +450,7 @@ var View = Backbone.Marionette.Layout.extend({
             }
 
 
-            if (values['brand'] === 'kvm') {
+            if (values['brand'] === 'kvm' || values['brand'] === 'bhyve') {
                 // disk size passed in as MiB.
                 values['disks'] = [
                     {'image_uuid': values['image_uuid']},
@@ -464,7 +464,7 @@ var View = Backbone.Marionette.Layout.extend({
 
         }
 
-        if ((values['brand'] === 'kvm' || values['brand'] === 'lx') && this.userKeys) {
+        if ((values['brand'] === 'kvm' || values['brand'] === 'bhyve' || values['brand'] === 'lx') && this.userKeys) {
             values.customer_metadata = {
                 root_authorized_keys: this.userKeys.map(function (k) {
                     return  _.str.trim(k).replace(/(\r\n|\n|\r)/gm, '');
-- 
2.21.0

